sdk/core/azure-core/azure/core/pipeline_client_async.py

```diff
@@ -107,19 +107,30 @@ def _request(self, method, url, params, headers, content, form_content, stream_c
 
         return request
 
+    def _format_url_section(self, template, **kwargs):
+        while True:
+            try:
+                return template.format(**kwargs)
+            except KeyError as key:
+                components = template.split("/")
+                new_components = [c for c in components if "{{{}}}".format(key.args[0]) not in c]
+                template = "/".join(new_components)
+
```

Comment:
This is not async itself, and duplicates the `_format_url_section` above. Can this be shared somehow?